toffy/normalize_test.py

```diff
@@ -0,0 +1,196 @@
+def test_fit_calibration_curve(plot_fit, obj_func):
+    x_vals = np.random.rand(15)
+    y_vals = np.random.rand(15)
+    _ = normalize.fit_calibration_curve(x_vals, y_vals, obj_func, plot_fit)
+
+
+@parametrize('obj_func, num_params', [('poly_2', 3), ('poly_3', 4), ('poly_4', 5),
+                                      ('poly_5', 6), ('log', 2), ('exp', 4)])
+def test_create_prediction_function(obj_func, num_params):
+    weights = np.random.rand(num_params)
+    pred_func = normalize.create_prediction_function(obj_func, weights)
+
+    _ = pred_func(np.random.rand(10))
+
+
+def test_combine_run_metrics():
+    with tempfile.TemporaryDirectory() as temp_dir:
+
+        # create full directory of files
+        bins = []
+        metrics = []
+        for i in range(1, 5):
+            bin_name = 'example_{}.bin'.format(i)
+            bins.append(bin_name)
+            metric_name = 'example_metric_{}.csv'.format(i)
+            metric_values = {'column_1': np.random.rand(10),
+                             'column_2': np.random.rand(10),
+                             'column_3': np.random.rand(10)}
+            metrics.append([metric_name, metric_values])
+
+        for bin_file in bins:
+            _make_blank_file(temp_dir, bin_file)
+
+        for metric in metrics:
+            name, values_df = metric[0], pd.DataFrame(metric[1])
+            values_df.to_csv(os.path.join(temp_dir, name), index=False)
+
+        normalize.combine_run_metrics(temp_dir, 'example_metric')
+
+        combined_data = pd.read_csv(os.path.join(temp_dir, 'example_metric_combined.csv'))
+
+        assert np.array_equal(combined_data.columns, ['column_1', 'column_2', 'column_3'])
+        assert len(combined_data) == len(bins) * 10
+
+        # check that previously generated combined file is removed with warning
+        with pytest.warns(UserWarning, match='previously generated'):
+            normalize.combine_run_metrics(temp_dir, 'example_metric')
+
+        # check that files with different lengths raises error
+        name, bad_vals = metrics[0][0], pd.DataFrame(metrics[0][1])
+        bad_vals = bad_vals.loc[0:5, :]
+        bad_vals.to_csv(os.path.join(temp_dir, name), index=False)
+
+        with pytest.raises(ValueError, match='files are the same length'):
+            normalize.combine_run_metrics(temp_dir, 'example_metric')
+        os.remove(os.path.join(temp_dir, name))
+        os.remove(os.path.join(temp_dir, 'example_1.bin'))
+
+        # different number of bins raises error
+        os.remove(os.path.join(temp_dir, bins[3]))
+        with pytest.raises(ValueError, match='Mismatch'):
+            normalize.combine_run_metrics(temp_dir, 'example_metric')
+
+        # empty directory raises error
+        empty_dir = os.path.join(temp_dir, 'empty')
+        os.makedirs(empty_dir)
+
+        with pytest.raises(ValueError, match='No bin files'):
+            normalize.combine_run_metrics(empty_dir, 'example_metric')
```

Comment:
I imagine these could be done via cases, but since we probably just need to crank this pr out now, this is fine.

Reply:
Is there an easy way to reuse the same data across different cases? When I tried to define variables within the `__init__` i got an error message. If so, I agree that this makes sense. However, it looks like I'd have to copy the code that makes the base case over in each of these, and then separately delete the data, which would clean up this test but just move the complexity to the cases file

Reply:
Yeah you can't specify anything in the `__init__` method (bummer).

The way I'd go about this is (kind-of) exemplified in `ark.utils.google_drive_utils_test`.  There's a `_fill_gdrive` method, as well as a `local_gdrive` decorator, which basically puts all tests into a pre-specified environment.  In the future 'various-utils' repo (which can include common test-utils), we can generalize this type of decorator to take a dictionary argument, which creates the specified environment (folders, files, etc.). 

For this purpose, a similar environment specifying decorator (or context class (using `with` keyword) ) can be used with the case class methods, as well as the `xfail` calls, and any parametrization.

Again, I'd be fine leaving it as is for now, but that's how I see file-i/o related testing being handled in the future

Reply:
Sounds good. I converted to a case so that we can expand in the future.